### PR TITLE
Enable amalgamation of lz4hc.c and lz4.c

### PR DESCRIPTION
**Pull request by antinucleon**
Repro:
```
$ cat lz4hc.c >> lz4.c
$ cc -c lz4.c   # compile properly
```

### CODE REVIEW

**Comment by antinucleon** [comment]
@Cyan4973 Done.

File: tests/Makefile [hunk 1]
```diff
@@ -152,11 +152,16 @@ endif
...
+test: test-lz4 test-lz4c test-frametest test-fullbench test-fuzzer test-install test-amalgamation
 
 test32: CFLAGS+=-m32
 test32: test
 
+test-amalgamation: $(LZ4DIR)/lz4.c $(LZ4DIR)/lz4hc.c
+	cat $(LZ4DIR)/lz4.c > lz4_all.c
+	cat $(LZ4DIR)/lz4hc.c >> lz4_all.c
+	$(CC) -I$(LZ4DIR) -c lz4_all.c
+
```
Comment (Cyan4973) [inline tests/Makefile]: nit: 
`$(RM) lz4_all.c`
(the `.o` will be removed by `make clean`, but `lz4_all.c` will not)

Comment (antinucleon) [reply tests/Makefile]: fixed

**Comment by Cyan4973** [comment]
Excellent ! Thanks @antinucleon !
Just waiting for end of CI tests before merge

**Pull request merged**